src/raftstore/store/peer.rs

```diff
@@ -464,6 +464,35 @@ impl Peer {
         self.raft_group.raft.state == StateRole::Leader
     }
 
+    // When this peer is the leader of the region before split,
+    // `accelerate_campaign_ticks` specifies the tick number to be accelerated
+    // after the region split. So this peer of new split region may campaign and become leader
+    // earlier than other follower peers.
+    pub fn accelerate_campaign_ticks(&self) -> usize {
+        // The best case for the fast campaign after a split is:
+        //   1. The leader of old region commits the log entry which contains the corresponding
+        //      split request.
+        //   2. The leader of old region replicates this log entry at 1 by sending AppendEntries
+        //      requests to the followers.
+        //   3. The leader of old region applies this log entry at 1, and then raftstore
+        //      accelerates ticks for the peer of the new region. This peer shares the same store
+        //      with old leader. It will start to campaign soon and send out RequestVote requests.
+        //   4. The followers receive AppendEntries requests from leader for replication in 2,
+        //      and then make it persistent and applie the split log entry so that
+        //      create a new region.
+        //   5. The followers receive RequestVote requests from leader in 3, and then vote
+        //      the campaign.
+        // However it's possible that 3 happens before 2, if RequestVote requests in 3 is sent
+        // to the followers earlier than AppendEntries requests in 2, the followers would not
+        // be ready to vote the campaign as 5 describes. Then the tick acceleration in 3 would be
+        // useless, so the campaign still happens after the election timeout.
+        // To make the best case happen at high degree of possibility, a time gap of
+        // `heartbeat_timeout * 2` is reserved for the followers to receive AppendEntries request
+        // and to apply them.
+        self.raft_group.raft.randomized_election_timeout -
```

Comment:
Why not using self.raft_group.raft.randomized_election_timeout - 1 directly so we can send vote after one tick (but now this will be delayed 1s, seem a little long).
I think we can even use self.raft_group.raft.randomized_election_timeout to send vote in next raft ready.